src/pint/utils.py

```diff
@@ -1265,6 +1265,98 @@ def split_swx(model, time):
+    frequency calculation. Sine and Cosine amplitudes are initially set to zero
+
+    User specifies either freqs or n_freqs. This function assumes that the timing model does not already
+    have any WaveX components. See add_wavex_component() or add_wavex_components to add WaveX components
+    to an existing WaveX model.
+
+    Parameters
+    ----------
+    fitter: PINT fitter object, None
+    freqs: iterable of float or astropy.quantity.Quantity
+        User inputed base frequencies
+    n_freqs: int, None
+        Number of wave frequencies to calculate using the equation: freq_n = 2 * pi * n / T_span
+        Where n is the wave number, and T_span is the total time span of the toas in the fitter object
+    """
+    from pint.models.wavex import WaveX
+
+    if (freqs is None) and (n_freqs is None):
+        raise ValueError(
+            "WaveX component base frequencies are not specified. "
+            "Please input either freqs or n_freqs"
+        )
+
+    if (freqs is not None) and (n_freqs is not None):
+        raise ValueError(
+            "Both freqs and n_freqs are specified. Only one or the other should be used"
+        )
+
+    if n_freqs == 0:
+        raise ValueError("Must use a non-zero number of wave frequencies")
+    fitter.model.add_component(WaveX())
+    T_span = fitter.toas.get_mjds().max() - fitter.toas.get_mjds().min()
+    nyqist_freq = 1.0 / (2.0 * T_span)
+    if freqs is not None:
+        if len(freqs) == 1:
+            fitter.model.WXFREQ_0001.value = freqs
+        else:
+            np.array(freqs)
+            freqs.sort()
+            if min(np.diff(freqs)) < nyqist_freq.value:
+                raise ValueError(
+                    "Wave frequency spacing is finer than frequency resolution of data"
+                )
+            fitter.model.WXFREQ_0001.value = freqs[0]
+            fitter.model.components["WaveX"].add_wavex_components(freqs[1:])
+
+    if n_freqs is not None:
+        if n_freqs == 1:
+            wave_freq = 2.0 * np.pi / T_span
+            fitter.model.WXFREQ_0001.value = wave_freq
+        else:
+            wave_numbers = np.arange(1, n_freqs + 1)
+            wave_freqs = 2.0 * np.pi * wave_numbers / T_span
+            fitter.model.WXFREQ_0001.value = wave_freqs[0]
+            fitter.model.components["WaveX"].add_wavex_components(wave_freqs[1:])
+    return fitter.model.components["WaveX"].get_indices()
+
+
+def translate_wave_to_wavex(model):
+    from pint.models import WaveX
+    from copy import deepcopy
+
+    def translate_wave_freqs(om, k):
+        return (om * (k + 1)) / (2.0 * np.pi)
+
+    wave_names = [
+        "WAVE%d" % ii for ii in range(1, model.components["Wave"].num_wave_terms + 1)
+    ]
+    wave_terms = deepcopy(
+        [getattr(model.components["Wave"], name) for name in wave_names]
+    )
+    wave_om = deepcopy(model.components["Wave"].WAVE_OM.quantity)
```

Comment:
When just using quantities, you can get the results of a deepcopy by assigning `new = 1.0 * old`.  

Also, is the deepcopy here just so that the variables aren't deleted when you delete the original `Wave` instance?

Reply:
> Also, is the deepcopy here just so that the variables aren't deleted when you delete the original `Wave` instance?
Yes